src/sagemaker/image_uris.py

```diff
@@ -106,22 +123,41 @@ def _processor(instance_type, available_processors):
         family = instance_type.split(".")[1]
         processor = "gpu" if family[0] in ("g", "p") else "cpu"
 
-    if processor in available_processors:
-        return processor
-
-    raise ValueError(
-        "Unsupported processor type: {} (for {}). "
-        "Supported type(s): {}.".format(processor, instance_type, ", ".join(available_processors))
-    )
+    _validate_arg("processor", processor, available_processors)
+    return processor
 
 
-def _validate_py_version(py_version, available_versions, framework, fw_version):
+def _validate_py_version_and_set_if_needed(py_version, version_config):
     """Checks if the Python version is one of the supported versions."""
-    if py_version not in available_versions:
+    available_versions = version_config.get("py_versions", version_config.keys())
+
+    if len(available_versions) == 0:
+        if py_version:
+            logger.info("Ignoring unnecessary Python version: %s.", py_version)
+        return None
+
+    if py_version is None and len(available_versions) == 1:
+        logger.info("Defaulting to only available Python version: %s", available_versions[0])
+        return available_versions[0]
+
+    _validate_arg("Python version", py_version, available_versions)
+    return py_version
+
+
+def _validate_arg(arg_name, arg, available_options):
+    """Checks if the arg is in the available options, and raises a ``ValueError`` if not."""
+    if arg not in available_options:
         raise ValueError(
-            "Unsupported Python version for {} {}: {}. You may need to upgrade "
-            "your SDK version (pip install -U sagemaker) for newer versions. "
-            "Supported Python version(s): {}.".format(
-                framework, fw_version, py_version, ", ".join(available_versions)
-            )
+            "Unsupported {arg_name}: {arg}. You may need to upgrade your SDK version "
+            "(pip install -U sagemaker) for newer {arg_name}s. Supported {arg_name}(s): "
+            "{options}.".format(arg_name=arg_name, arg=arg, options=", ".join(available_options))
         )
+
+
+def _format_tag(version, processor, py_version):
+    """Creates a tag for the image URI."""
+    tag = "{}-{}".format(version, processor)
+    if py_version:
+        tag += "-{}".format(py_version)
```

Comment:
re: [this](https://github.com/aws/sagemaker-python-sdk/pull/1707/files#diff-6854db2a7b2e68804b6b3f8d813b345cR51-R53), unify the way you do this?